orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -846,24 +846,35 @@ void addObjectMethodsForTraining(py::module& m, ExecutionProviderRegistrationFn
               const std::vector<OrtValue>& user_inputs, std::vector<OrtValue>& user_outputs) -> void {
              ORT_THROW_IF_ERROR(model->EvalStep(user_inputs, user_outputs));
            })
-      .def("reset_grad", [](onnxruntime::training::api::Module* model) -> void {
-        ORT_THROW_IF_ERROR(model->ResetGrad());
-      })
-      .def("copy_parameters_to_buffer", [](onnxruntime::training::api::Module* model, OrtValue& output) -> void {
-        ORT_THROW_IF_ERROR(model->CopyParametersToBuffer(output));
-      })
-      .def("copy_buffer_to_parameters", [](onnxruntime::training::api::Module* model, OrtValue& input) -> void {
-        ORT_THROW_IF_ERROR(model->CopyBufferToParameters(input));
-      })
-      .def("get_parameters_size", [](onnxruntime::training::api::Module* model, bool trainable_only) -> size_t {
-        return model->GetParametersSize(trainable_only);
-      })
-      .def("save_checkpoint", [](onnxruntime::training::api::Module* model, const std::string& checkpoint_path) -> void {
-        onnxruntime::training::api::CheckpointState state;
-        ORT_THROW_IF_ERROR(model->GetStateDict(state.module_checkpoint_state));
-        ORT_THROW_IF_ERROR(onnxruntime::training::api::SaveCheckpoint(state,
-                                                                      ToPathString(checkpoint_path)));
-      });
+      .def("reset_grad",
+           [](onnxruntime::training::api::Module* model) -> void {
+             ORT_THROW_IF_ERROR(model->ResetGrad());
+           })
+      .def("copy_parameters_to_buffer",
+           [](onnxruntime::training::api::Module* model, OrtValue& output) -> void {
+             ORT_THROW_IF_ERROR(model->CopyParametersToBuffer(output));
+           })
+      .def("copy_buffer_to_parameters",
+           [](onnxruntime::training::api::Module* model, OrtValue& input) -> void {
+             ORT_THROW_IF_ERROR(model->CopyBufferToParameters(input));
+           })
+      .def("get_parameters_size",
+           [](onnxruntime::training::api::Module* model, bool trainable_only) -> size_t {
+             return model->GetParametersSize(trainable_only);
+           })
+      .def("save_checkpoint",
+           [](onnxruntime::training::api::Module* model, const std::string& checkpoint_path) -> void {
+             onnxruntime::training::api::CheckpointState state;
+             ORT_THROW_IF_ERROR(model->GetStateDict(state.module_checkpoint_state));
+             ORT_THROW_IF_ERROR(onnxruntime::training::api::SaveCheckpoint(state,
+                                                                           ToPathString(checkpoint_path)));
+           })
+      .def("export_model_for_inferencing",
+           [](onnxruntime::training::api::Module* model, const std::string& inference_model_path,
+              const std::vector<std::string>& graph_output_names) -> void {
+             ORT_THROW_IF_ERROR(model->ExportModelForInferencing(inference_model_path,
```

Comment:
looks we never check the nullptr for model, do you think that is a expected practice here?

Reply:
I think it is safe to not check. pybind can take care of that for us.

```py
module= C.Module(...)
module.export_model_for_inferencing(...)
```
so, we should expect that the first argument to this function is not a nullptr. But i think adding a check would be good.